### PR TITLE
TBQ-4: enroll like provider

### PR DESCRIPTION
**Pull request by IvanKovalov**

### CODE REVIEW

File: src/main/java/ua/shpp/eqbot/command/CommandName.java [hunk 1]
```diff
@@ -9,7 +9,7 @@ public enum CommandName {
 
     CHANGE_ROLE_TO_PROVIDER("Change role to Provider"),
 
-    REGISTR_NEW_PROVIDER("Реєстрація нового провайдера"),
+    REGISTR_NEW_PROVIDER("реєстрація нового провайдера"),
```
Comment (Sekator778) [inline src/main/java/ua/shpp/eqbot/command/CommandName.java]: change

File: src/main/java/ua/shpp/eqbot/commandchain/changerole/RegistrationProviderChain.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+import ua.shpp.eqbot.repository.ProvideRepository;
+import ua.shpp.eqbot.service.SendBotMessageService;
+
+import java.util.ArrayList;
+
+public class RegistrationProviderChain implements CommandChain {
+
+    private final static Logger LOGGER = LoggerFactory.getLogger(RegistrationProviderChain.class);
+    private final ArrayList<Command> ChangeRoleChain = new ArrayList<>();
+    private int i = (-1);
```
Comment (Sekator778) [inline src/main/java/ua/shpp/eqbot/commandchain/changerole/RegistrationProviderChain.java]: add comment plz
what this number

File: src/main/java/ua/shpp/eqbot/command/CommandName.java [hunk 3]
```diff
@@ -10,7 +10,7 @@ public enum CommandName {
...
```
Comment (Tsurkan) [inline src/main/java/ua/shpp/eqbot/command/CommandName.java]: Why small?

File: src/main/java/ua/shpp/eqbot/model/ProviderEntity.java [hunk 4]
```diff
@@ -0,0 +1,69 @@
+package ua.shpp.eqbot.model;
+
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import java.util.Objects;
+
+@Entity
```
Comment (Tsurkan) [inline src/main/java/ua/shpp/eqbot/model/ProviderEntity.java:9]: Will there be a connection to a regular user?

File: src/main/java/ua/shpp/eqbot/commandchain/changerole/RegistrationProviderChain.java (hunk 2 above)
Comment (Tsurkan) [inline src/main/java/ua/shpp/eqbot/commandchain/changerole/RegistrationProviderChain.java]: Is it possible in another way?

File: src/main/java/ua/shpp/eqbot/telegrambot/EqTelegramBot.java [hunk 5]
```diff
@@ -43,8 +52,14 @@ public EqTelegramBot(UserRepository userRepository, ServiceRepository serviceRep
     public void onUpdateReceived(Update update) {
         if (update.hasCallbackQuery()) {
             callbackQueryHandler(update);
+        }else if(update.getMessage().getText().equals("Change role to Provider")){
```
Comment (Tsurkan) [inline src/main/java/ua/shpp/eqbot/telegrambot/EqTelegramBot.java]: Perform by separate methods?

File: src/main/java/ua/shpp/eqbot/commandchain/changerole/RegistrationProviderStep.java [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+    REG_ADD_CITY_STEP,
+    REG_DONE;
+    public RegistrationProviderStep next (RegistrationProviderStep registrationProviderStep){
+        switch (registrationProviderStep){
+            case REG_MESSAGE_STEP:
+                return REG_ADD_NAME_STEP;
+            case REG_ADD_NAME_STEP:
+                return REG_ADD_CITY_STEP;
+            case REG_ADD_CITY_STEP:
+                return REG_DONE;
```
Comment (Kornilchenko) [inline src/main/java/ua/shpp/eqbot/commandchain/changerole/RegistrationProviderStep.java:15]: case REG_ADD_CITY_STEP:
                return REG_DONE;
            default:
                return REG_DONE;
replace with
case REG_ADD_CITY_STEP:
default:
                return REG_DONE;

**Comment by Sekator778** [comment]
ok
I think it should be checked before internationalization

**Review by Sekator778 — APPROVED** [review]
I think it should be checked before internationalization

File: src/main/java/ua/shpp/eqbot/commandchain/changerole/RegistrationProviderChain.java (hunk 2 above)
Comment (Tsurkan) [reply src/main/java/ua/shpp/eqbot/commandchain/changerole/RegistrationProviderChain.java]: So we leave -1?

Comment (IvanKovalov) [reply src/main/java/ua/shpp/eqbot/commandchain/changerole/RegistrationProviderChain.java]: Там я змінив всю реалізацію ще сьогодні зранку. В пул реквесті нова логіка. Можете глянути

**Review by 1ulans1 — APPROVED** [review]

**Review by Kornilchenko — APPROVED** [review]

**Pull request merged**